pkg/traces/traces.go

```diff
@@ -20,6 +21,16 @@ const (
 	defaultAttributeNamespace = "unknown"
 )
 
+// StartHttpRequestSpan returns a copy of the request with a new context to include span info and span,
+// including information about the calling function as appropriate.
+// Standardizes the tracer name. The caller is always responsible for
+// ending the span. `keyVals` should be a list of pairs, where the first in the pair is a
+// string representing the attribute key and the second in the pair is the attribute value.
+func StartHttpRequestSpan(r *http.Request, keyVals ...interface{}) (*http.Request, trace.Span) {
+	ctx, span := StartSpan(r.Context(), keyVals...)
+	return r.WithContext(ctx), span
```

Comment:
I'm mostly sure this is the correct way to do this. I followed the otel code [here](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/instrumentation/net/http/otelhttp/v0.46.0/instrumentation/net/http/otelhttp/handler.go#L229), but I would not mind a second pair of eyes to make sure I'm doing it correctly. We could also use the otel included https stuff, but maybe it's over kill for our little servers?